### PR TITLE
Implemented order logic. Adjusted all fetch types and created custom …

### PR DESCRIPTION
**Pull request by Vitalii-Kazymyrovych**
…repository methods.

### CODE REVIEW

File: src/main/java/com/project/bookstore/model/Order.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Table(name = "orders")
+@Setter
+@Getter
+@RequiredArgsConstructor
+public class Order {
```
Comment (koval-kn) [inline src/main/java/com/project/bookstore/model/Order.java:29]: dont forget about soft delete approach

File: src/main/java/com/project/bookstore/model/OrderItem.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Table(name = "order_items")
+@Setter
+@Getter
+@RequiredArgsConstructor
+public class OrderItem {
```
Comment (koval-kn) [inline src/main/java/com/project/bookstore/model/OrderItem.java:23]: dont forget about soft delete approach

File: src/main/java/com/project/bookstore/model/OrderItem.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Table(name = "order_items")
+@Setter
+@Getter
+@RequiredArgsConstructor
```
Comment (koval-kn) [inline src/main/java/com/project/bookstore/model/OrderItem.java]: redundant
```suggestion
```

File: src/main/java/com/project/bookstore/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Set;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Table(name = "orders")
+@Setter
+@Getter
+@RequiredArgsConstructor
```
Comment (koval-kn) [inline src/main/java/com/project/bookstore/model/Order.java]: redundant
```suggestion
```

File: src/main/java/com/project/bookstore/model/Order.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        private final String name;
+
+        Status(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public String toString() {
+            return name;
+        }
```
Comment (koval-kn) [inline src/main/java/com/project/bookstore/model/Order.java]: enum.name().toLowerCase() can be used instead
```suggestion
        NEW,
        PAID,
        PROCESSING,
        DELIVERED,
        CANCELLED
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**